node/src/accountant/mod.rs

```diff
@@ -4089,4 +4132,87 @@ mod tests {
+        let mut subject = AccountantBuilder::default().build();
+        subject.logger = Logger::new(test_name);
+        match message.scan_type {
+            ScanType::Payables => subject.scanners.payable.mark_as_started(SystemTime::now()),
+            ScanType::PendingPayables => subject
+                .scanners
+                .pending_payable
+                .mark_as_started(SystemTime::now()),
+            ScanType::Receivables => subject
+                .scanners
+                .receivable
+                .mark_as_started(SystemTime::now()),
+        }
+        let subject_addr = subject.start();
+        let system = System::new("test");
+        let peer_actors = peer_actors_builder().ui_gateway(ui_gateway).build();
+        subject_addr.try_send(BindMessage { peer_actors }).unwrap();
+
+        subject_addr.try_send(message.clone()).unwrap();
+
+        subject_addr
+            .try_send(AssertionsMessage {
+                assertions: Box::new(move |actor: &mut Accountant| {
+                    let is_scan_running = match message.scan_type {
+                        ScanType::Payables => actor.scanners.payable.scan_started_at(),
+                        ScanType::PendingPayables => {
+                            actor.scanners.pending_payable.scan_started_at()
+                        }
+                        ScanType::Receivables => actor.scanners.receivable.scan_started_at(),
+                    };
+                    assert_eq!(is_scan_running, None);
+                }),
+            })
+            .unwrap();
+        System::current().stop();
+        system.run();
+        let ui_gateway_recording = ui_gateway_recording_arc.lock().unwrap();
+        match message.response_skeleton_opt {
+            Some(response_skeleton) => {
+                let expected_message_sent_to_ui = NodeToUiMessage {
+                    target: ClientId(response_skeleton.client_id),
+                    body: MessageBody {
+                        opcode: "scan".to_string(),
+                        path: MessagePath::Conversation(response_skeleton.context_id),
+                        payload: Err((
+                            SCAN_ERROR,
+                            format!("{:?} scan failed: '{}'", message.scan_type, message.msg),
+                        )),
+                    },
+                };
+                assert_eq!(
+                    ui_gateway_recording.get_record::<NodeToUiMessage>(0),
+                    &expected_message_sent_to_ui
+                );
+                TestLogHandler::new().assert_logs_contain_in_order(vec![
+                    &format!("ERROR: {}: Received ScanError: {:?}", test_name, message),
+                    &format!(
+                        "ERROR: {}: Sending UiScanResponse: {:?}",
+                        test_name, expected_message_sent_to_ui
+                    ),
+                ]);
+            }
+            None => {
+                let message_sent_to_ui_opt =
```

Comment:
I think you can leave out these three lines and just assert that the recording length is zero.